File: day1/task1.zig
const std = @import("std");

pub fn main() !void {
  const filename = "input.txt";
  var file = try std.fs.cwd().openFile(filename, .{});
  defer file.close();

  var buffered = std.io.bufferedReader(file.reader());
  var reader = buffered.reader();

  var floor = @as(i32, 0);

  while (true) {
    const symbol = reader.readByte() catch |err| switch(err) {
      error.EndOfStream => break,
      else => return err,
    };
    switch (symbol) {
      '(' => { floor += 1; },
      ')' => { floor -= 1; },
      else => {},
    }
  }

  std.debug.print("Floor is: {}", .{floor});
}

File: day1/task1.jl
open("input.txt") do file
  floor :: Int32 = 0
  while !eof(file)
    symbol :: Char = read(file, Char)
    if symbol == '('
      floor += 1
    elseif symbol == ')'
      floor -= 1
    end
  end
  println(floor)
end
